src/net/client/mod.rs

```diff
@@ -102,31 +133,46 @@ impl MpcHelperClient {
                 ClientIdentity::None => None,
             };
             (
-                HttpsConnector::new_with_connector(make_http_connector()),
+                HttpsConnectorBuilder::new()
+                    .with_native_roots()
+                    .https_or_http()
+                    .enable_http2()
+                    .wrap_connector(make_http_connector()),
                 auth_header,
             )
         } else {
-            let mut builder = TlsConnector::builder();
-            if let Some(certificate) = peer_config.certificate {
-                builder
-                    .disable_built_in_roots(true)
-                    .add_root_certificate(Certificate::from_der(certificate.as_ref()).unwrap());
+            let builder = rustls::ClientConfig::builder().with_safe_defaults();
+            let client_config = if let Some(certificate) = peer_config.certificate {
+                let cert_store = {
+                    let mut store = RootCertStore::empty();
+                    store.add(&certificate).unwrap();
+                    store
+                };
+
+                let builder = builder.with_root_certificates(cert_store);
                 match identity {
-                    ClientIdentity::Certificate(identity) => {
-                        builder.identity(identity);
-                    }
+                    ClientIdentity::Certificate((cert, pk)) => builder
+                        .with_single_cert(vec![cert], pk)
```

Comment:
I only just noticed this, but there is a potential risk here in that a PKCS\#8 file containing a certificate path (i.e., more than just the end-entity certificate) is not used.  I think that the right answer is to retain `Vec<Certificate>` instead.  The trust store in `cert_store` is fine (the end-entity certificate is all we need for that).